students/M_Sunday/lesson08/circle.py

```diff
@@ -0,0 +1,78 @@
+#!/usr/bin/python
+from math import pi
+
+
+class Circle(object):
+
+    @staticmethod
+    def entry_check(the__radius):
+        if isinstance(the__radius, (str, list, tuple, dict)):
+            raise TypeError("Radius/Diameter entry must be a single, positive, and "
+                            "non-string value")
+        else:
+            if the__radius < 0:
+                raise ValueError("Radius/Diameter must be greater than 0")
+            else:
+                return the__radius
+
+    def __init__(self, the_radius):
+        self._radius = self.entry_check(the_radius)
```

Comment:
I don't think you need to call entry_check here since you're calling it in the setter